src/app/hooks/useMaintenance.ts

```diff
@@ -0,0 +1,30 @@
+import { useSelector, useDispatch } from 'react-redux';
+import { useInjectReducer, useInjectSaga } from 'utils/redux-injectors';
+import {
+  actions,
+  maintenanceSlice,
+  reducer as maintenanceReducer,
+} from 'store/global/maintenance-store/slice';
+import { selectMaintenanceState } from 'store/global/maintenance-store/selectors';
+import { maintenanceStateSaga } from 'store/global/maintenance-store/saga';
+import { useEffect } from 'react';
+
+export function useMaintenance() {
+  useInjectReducer({ key: maintenanceSlice, reducer: maintenanceReducer });
+  useInjectSaga({ key: maintenanceSlice, saga: maintenanceStateSaga });
+  const dispatch = useDispatch();
+  const store = useSelector(selectMaintenanceState);
+
+  useEffect(() => {
+    if (!store || Object.keys(store).length === 0) {
+      dispatch(actions.fetchMaintenanceState());
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
Thinking that maybe we should refetch state from the backend every minute or so, otherwise users who already in the dapp will be able to make transactions and pay fee for failed transactions.

You can do it in saga:

```typescript
import delay from '@redux-saga/delay-p'

function* fetchStatePeriodicallySaga() {
  while (true) {
    yield call(callCreateMaintenanceState);
    yield delay(2000);
  }
}
```